client/src/components/base/ItemFilterBar/ItemFilterBar.tsx

```diff
@@ -2,30 +2,30 @@ import React, { useState } from 'react';
 import styled from '@emotion/styled';
 import { baemin, baeminFont, baeminThickFont, lightBlack } from '@/static/style/common';
 
-const filterName = ['추천순', '인기순', '최신순', '낮은가격순', '높은가격순'];
+const filterName: string[] = ['추천순', '인기순', '최신순', '높은가격순', '낮은가격순'];
 
 type ItemFilterBarProps = {
   handleFilter: (num: number) => void;
   totalProductCount: number;
 };
 
 const ItemFilterBar = ({ handleFilter, totalProductCount }: ItemFilterBarProps) => {
-  const [index, setIndex] = useState(-1);
+  const [index, setIndex] = useState<number>(0);
 
-  const onClickFilter = (e: React.MouseEvent) => {
-    const { target } = e;
-    if (!(target instanceof HTMLElement)) return;
-    setIndex(Number(target.dataset.idx));
-    handleFilter(Number(target.dataset.idx));
+  const handleClickFilter = (e: React.MouseEvent<HTMLLIElement>) => {
```

Comment:
이 부분인 너무 길다 싶으시면
`import React, {MouseEvent} from 'react'` 하신다음 

```suggestion
  const handleClickFilter = (e: MouseEvent<HTMLLIElement>) => {
```
와 같이 하셔도 될 듯 합니다!! 👍

Reply:
오호 참고하겠습니다 👍